lib/host.js

```diff
@@ -305,7 +335,47 @@ HostConnectionPool.prototype.forceShutdown = function () {
   });
 };
 
+/**
+ * Closes all open connections and schedules reconnection
+ * @param {Number} delay
+ */
+HostConnectionPool.prototype.scheduleReconnection = function (delay) {
+  this._forceShutdown();
+  this.reconnectionTimeout = setTimeout(this._attemptReconnection.bind(this), delay);
+};
+
+/**
+ * Prevents reconnection timeout from triggering
+ */
+HostConnectionPool.prototype.clearReconnection = function () {
+  if (!this.reconnectionTimeout) {
+    return;
+  }
+  clearTimeout(this.reconnectionTimeout);
+  this.reconnectionTimeout = null;
+};
+
+HostConnectionPool.prototype._attemptReconnection = function () {
+  this.reconnectionTimeout = null;
+  var c = this._createConnection();
+  var self = this;
+  c.open(function (err) {
+    if (err) {
+      self.log('info', util.format('Reconnection attempt to host %s failed', self.address), err);
+      self.host.setDown();
```

Comment:
Ah, it's not that simple, since we still need to schedule another reconnection.  We need to tune the logic such that the 'down' event is not emitted in this case but we still attempt to a reconnect.